File: codetree/프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {        
        int[] stud_1 = {1,2,3,4,5};
        int[] stud_2 = {2,1,2,3,2,4,2,5};
        int[] stud_3 = {3,3,1,1,2,2,4,4,5,5};
        int solved_1 = 0;
        int solved_2 = 0;
        int solved_3 = 0;
        
        for (int i=0; i<answers.length; i++) {
            if (answers[i]==stud_1[i%stud_1.length]) {
                solved_1++;
            }
            if (answers[i]==stud_2[i%stud_2.length]) {
                solved_2++;
            }
            if (answers[i]==stud_3[i%stud_3.length]) {
                solved_3++;
            }
        }
        
        int mxScore = Math.max(solved_1, Math.max(solved_2, solved_3));
        //System.out.println(mxScore);
        List<Integer> answer = new ArrayList<>();
        
        if (solved_1==mxScore) {
            answer.add(1);
        }
        if (solved_2==mxScore) {
            answer.add(2);
        }
        if (solved_3==mxScore) {
            answer.add(3);
        }
        
        return answer.stream().mapToInt(i->i).toArray();
    }
}

File: codetree/프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    st1 = [1, 2, 3, 4, 5]
    st2 = [2, 1, 2, 3, 2, 4, 2, 5]
    st3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    cnt = [0]*3 # 1~3번 학생 정답 맞힌 개수
    for i in range(len(answers)):
        # print(st1[i%len(st1)])
        if answers[i]==st1[i%len(st1)]:
            cnt[0]+=1
        
        if answers[i]==st2[i%len(st2)]:
            cnt[1]+=1
        
        if answers[i]==st3[i%len(st3)]:
            cnt[2]+=1
    
    mx_cnt = max(cnt)
    
    for i in range(3):
        if cnt[i]==mx_cnt:
            answer.append(i+1)
    return answer